src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -130,44 +152,123 @@ private String convertToValidCountryOfOrigin(String countryCode) {
         return isoCountry;
     }
 
-    private String getDefaultSupplierActiveValue() {
-        return getParameterValueString(CUPurapParameterConstants.JAGGAER_DEFAULT_SUPPLIER_ACTIVE_VALUE);
-    }
-
     private JaggaerBasicValue buildJaggerLegalStructure(VendorDetail detail) {
         String kfsOwnerShipCode = detail.getVendorHeader().getVendorOwnershipCode();
         JaggaerLegalStructure legalStructure = JaggaerLegalStructure
                 .findJaggaerLegalStructureByKfsOwnershipCode(kfsOwnerShipCode);
         return JaggaerBuilder.buildJaggaerBasicValue(legalStructure.jaggaerLegalStructureName);
     }
+    
+    protected void processWebsiteUrl(VendorDetail detail, Supplier supplier) {
+        if (StringUtils.isNotBlank(detail.getVendorUrlAddress())) {
+            if (isValidUrl(detail.getVendorUrlAddress(), detail.getVendorNumber())) {
+                supplier.setWebSiteURL(JaggaerBuilder.buildJaggaerBasicValue(detail.getVendorUrlAddress()));
+            } else {
+                String websiteErrorMessage  = MessageFormat.format(
+                        configurationService.getPropertyValueAsString(CUPurapKeyConstants.JAGGAER_XML_WEBSITE_ERROR),
+                        detail.getVendorNumber(), detail.getVendorUrlAddress());
+                supplier.getJaggaerUploadSupplierVendorDetailDto().getNotes().add(websiteErrorMessage);
+            }
+        }
+    }
+    
+    protected boolean isValidUrl(String url,  String vendorNumber) {
+        LOG.debug("isValidUrl, entering with url '{}' and vendor number '{}'", url, vendorNumber);
+        if (StringUtils.isNotBlank(url) && !numberPattern.matcher(url).matches()) {
+            try {
+                new URL(url).toURI();
+                return true;
+            } catch (URISyntaxException | MalformedURLException e) {
+                LOG.debug("isValidUrl, failed validation by java.net.URI for vendor {} with URL of {}", vendorNumber, url);
+            }
+            
+            try {
+                InetAddress address = InetAddress.getByName(url);
+                LOG.debug("isValidUrl, for url {} the host address is {}", url, address.getHostAddress());
+                if (StringUtils.isNotBlank(address.getHostAddress())) {
+                    return true;
+                }
+            } catch (UnknownHostException e) {
+                LOG.debug("isValidUrl, failed validation by java.net.InetAddress for vendor {} with URL of {}", vendorNumber, url);
+            }
+        }
+        LOG.error("isValidUrl, vendor number '{}' has an invalid URL of '{}'", vendorNumber, url);
+        return false;
+    }
```

Comment:
Good question, I was debating that while coding this.    
The LOG.error statement will display what the website URL is in KFS and the vendor number it is associated with.  That will be the critical detail to debugging this if needed.  The to be written user report will contain a note if an invalid website URL is hit.
I didn't want to regularly log in the catch statement because I thought it would just add clutter to job LOG files.  if the first try block will fail if isn't a properly formatted address, like no HTTP.  The second block will fail if there are no DNS entries for what is provided.  I can switch these to INFO if you think it would be helpful.